### ISSUE TITLE
Tofu appears in UTF-8-encoded Markdown README

### ISSUE THREAD

**Issue by lofidevops** [pypi/warehouse#12119]
**Describe the bug**

My README.md contains non-ASCII characters. [When rendered on PyPI](https://pypi.org/project/samosa/) they appear as [tofu](https://en.wiktionary.org/wiki/tofu#Noun).

**Expected behavior**

These characters should be rendered normally.

**To Reproduce**

 * See [README.md](https://gitlab.com/lofidevops/samosa/-/blob/main/README.md), [pyproject.toml](https://gitlab.com/lofidevops/samosa/-/blob/main/pyproject.toml)
 * Run `pdm publish`

**My Platform**

 * Occurs in Firefox, GNOME Web and Chrome. I do not see tofu on other pages (link below), so I don't think this is a browser issue.
 * PDM 2.1.2, installed via pipx 1.0.0, running Python 3.10.4

**Additional context**

 * On my machine `file --mime README.md` returns `README.md: text/plain; charset=utf-8` so I believe the file encoding is correct before upload.
 * I believe I have declared the [readme field](https://packaging.python.org/en/latest/specifications/declaring-project-metadata/#readme) correctly, see `pyproject.toml` above.
 * READMEs for other packages render correctly, for example [epitran](https://pypi.org/project/epitran/) contains a variety of non-ASCII characters for various alphabets.

**Comment by di** [comment]
This is likely an issue with pdm. The raw value that PyPI was provided for the [`Description`](https://packaging.python.org/en/latest/specifications/core-metadata/#description) metadata field includes TOFU:

```
warehouse=> select raw from release_descriptions where id = 'e092febc-d162-4e7d-80fd-6565d2d24201';
                                                                                raw
--------------------------------------------------------------------------------------------------------------------------------------------------------------------
 # Samosa (���������������)                                                                                                                                        +
                                                                                                                                                                   +
 Enforce a triangular Git workflow. If this is not possible, explain why.                                                                                          +
```

However it's clear that the actual value in the distribution does not contain TOFU: https://inspector.pypi.io/project/samosa/0.1.2/packages/f5/02/d6fa62fc59eca0d52c4043e215ec4ee7f0590ae6592403e2b211c670cae3/samosa-0.1.2.tar.gz/samosa-0.1.2/PKG-INFO#line.10

Because PyPI does zero introspection of the artifact and depends on the metadata provided at upload time instead, this means that when the uploader was preparing the metadata to provide along with the artifact, it incorrectly read this field from the distribution artifact.

Can you confirm that a) that you're using the latest version of pdm and b) that this issue does not occur when you publish with https://pypi.org/p/twine/ instead?

**Comment by di** [comment]
(Actually, I can determine that you used pdm 2.1.2, which is the current latest)

**Comment by lofidevops** [comment]
Thanks for the info! This was indeed a PDM bug, now fixed in 2.1.3 https://github.com/pdm-project/pdm/commit/7ade7625194fa99596551f43782e2c43c3099cc5
